src/lib.rs

```diff
@@ -66,15 +66,28 @@ impl<'a, 'b> DrawingBackend for RatatuiBackend<'a, 'b> {
         coord: BackendCoord,
         radius: u32,
         style: &S,
-        _fill: bool,
+        fill: bool,
     ) -> std::result::Result<(), DrawingErrorKind<Self::ErrorType>> {
         let (x, y) = backend_to_canvas_coords(coord, self.size);
-        self.canvas.draw(&canvas::Circle {
-            x,
-            y,
-            radius: radius.into(),
-            color: convert_color(style.color()),
-        });
+        if fill {
+            for radius in 1..=radius {
```

Comment:
I am concerned that there may be empty pixels not covered by any circles due to rounding issues. What about we draw line by line with pyth them here?

```rs
for dy in -radius..=radius {
    let half_width = ((radius.pow(2) - dy.pow(2)) as f64).sqrt();
    self.draw_line(
        ((coord.0 - half_width) as i32, coord.1 + dy),
        ((coord.0 + half_width) as i32, coord.1 + dy),
        style,
    );
}
```

Reply:
Good catch.  Didn't show up in the small circles I tested in my application, but very much does at larger scales.  Your solution works well.

I've added a test rig and will add as a commit in my PR, and this demonstrates that unfilled circles don't work.  Do you see them working?

Reply:
yeah, I am planning to send a PR to ratatui to fix their implementation.

Reply:
See ratatui-org/ratatui#917